File: src/main/java/nz/co/aetheric/sameness/equatators/VoidEquatator.java
package nz.co.aetheric.sameness.equatators;

import nz.co.aetheric.sameness.Equatator;

/**
 * TODO: What is the purpose of this class?
 * <p>Author: <a href="http://gplus.to/tzrlk">Peter Cummuskey</a></p>
 */
public class VoidEquatator implements Equatator<Void> {

	public static VoidEquatator voidEquatator = new VoidEquatator();

	@Override
	public Class<Void> getType() {
		return Void.class;
	}

	@Override
	public <L extends Void, R extends Void> boolean isEquatable(L left, R right) {
		return true;
	}

}

File: src/main/kotlin/nz/co/aetheric/sameness/equatators/VoidEquatator.kt
package nz.co.aetheric.sameness.equatators

import nz.co.aetheric.sameness.Equatator

/**
 * TODO: What is the purpose of this class?
 *
 * Author: [Peter Cummuskey](http://gplus.to/tzrlk)
 */
class VoidEquatator : Equatator<Void> {

	companion object {
		val voidEquatator = VoidEquatator()
	}

	override val type: Class<Void>
		get() = Void::class.java

	override fun <L : Void, R : Void> isEquatable(left: L?, right: R?)
			= true

}
